src/platforms/PolygonId/components/PolygonIdMintingModalContent.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  ModalBody,
+  ModalHeader,
+  Skeleton,
+  Spacer,
+  Stack,
+  Text,
+} from "@chakra-ui/react"
+import useGuild from "components/[guild]/hooks/useGuild"
+import useUser from "components/[guild]/hooks/useUser"
+import { useState } from "react"
+import { Role } from "types"
+import MintableRole from "./MintableRole"
+import NoDID from "./NoDID"
+import PolygonIdQRCode from "./PolygonIdQRCode"
+
+const LoadingState = () => (
+  <>
+    <ModalHeader pb={0}>
+      <Text>Mint PolygonID proofs</Text>
+    </ModalHeader>
+    <ModalBody pt={8}>
+      <Stack gap={3}>
+        <Skeleton height={16}></Skeleton>
+        <Skeleton height={16}></Skeleton>
+      </Stack>
+    </ModalBody>
+  </>
+)
+
+const ErrorState = () => (
+  <Alert status="error" pb={5} alignItems={"center"} mb={5}>
+    <AlertIcon />
+    <Stack>
+      <AlertDescription position="relative" top={0.5} fontWeight="semibold" pr="4">
+        Server not available
+      </AlertDescription>
+    </Stack>
+    <Spacer />
+  </Alert>
+)
+
+const PolygonIdMintingModalContent = () => {
```

Comment:
This is not a too complex component, and I think we don't use it in multiple files, so in my opinion we can move this logic&markup into `MintPolygonIdProofModal`